core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1171,6 +1241,28 @@ wasm_runtime_deinstantiate_internal(WASMModuleInstanceCommon *module_inst,
 #endif
 }
 
+bool
+wasm_runtime_set_running_mode(wasm_module_inst_t module_inst,
+                              RunningMode running_mode)
+{
+#if WASM_ENABLE_FAST_JIT != 0 || WASM_ENABLE_JIT != 0
+    if (wasm_runtime_is_running_mode_supported(running_mode)) {
+        WASMModuleInstance *actual_module_inst =
+            (WASMModuleInstance *)module_inst;
+        actual_module_inst->e->running_mode = running_mode;
```

Comment:
1. Module instance can be AOTModuleInstance, should not convert to WASMModuleInstance directly.

2. Had better to convert running mode 0 to the actual running mode

Code may be like:
```C
{
#if WASM_ENABLE_AOT != 0
    if (module_inst->type == Wasm_Module_AoT)
        return true;
#endif

#if WASM_ENABLE_INTERP != 0
    if (module_inst->type == Wasm_Module_Bytecode) {
        WASMModuleInstance *module_inst_interp =
            (WASMModuleInstance *)module_inst;

        if (running_mode == 0) {
#if WASM_ENABLE_FAST_JIT == 0 && WASM_ENABLE_LLVM_JIT == 0
            running_mode = Mode_Interp;
#elif WASM_ENABLE_FAST_JIT != 0 && WASM_ENABLE_LLVM_JIT == 0
            running_mode = Mode_Fast_JIT;
#elif WASM_ENABLE_FAST_JIT == 0 && WASM_ENABLE_LLVM_JIT != 0
            running_mode = Mode_LLVM_JIT;
#elif WASM_ENABLE_FAST_JIT != 0 && WASM_ENABLE_LLVM_JIT != 0
            running_mode = Mode_Multi_Tier_JIT;
#endif
        }
        module_inst_interp->e->running_mode = running_mode;

        /* And here we should handle the function pointers, and call this API when instantiating the wasm module in wasm_runtime.c, so we don't need to modify the function pointers in wasm_interp_classic.c, let's do it later */
        return true;
    }
#endif

    return false;
}
```